panoptica/panoptica_evaluator.py

```diff
@@ -114,13 +118,18 @@ def _yaml_repr(cls, node) -> dict:
     @measure_time
     def evaluate(
         self,
-        prediction_arr: np.ndarray,
-        reference_arr: np.ndarray,
+        prediction_arr: Any,
+        reference_arr: Any,
```

Comment:
This is misleading, and should rather be to make the API explicit:
```suggestion
        prediction_arr: Union[str, sitk.Image, np.ndarray, torch.Tensor],
        reference_arr: Union[str, sitk.Image, np.ndarray, torch.Tensor],
```

Reply:
Right - in this case, I would suggest a few "hard" requirements (SimpleITK, for instance), and then another `Any`. Thoughts?

Reply:
I also admire the problem :(

Reply:
```
if TYPE_CHECKING:
    import torch
    import SimpleITK as sitk
    import nibabel as nib
    
[...]

@citation_reminder
    @measure_time
    def evaluate(
        self,
        prediction_arr: Union[np.ndarray, "torch.Tensor", "nib.nifti1.Nifti1Image", "sitk.Image"],
        reference_arr: Union[np.ndarray, "torch.Tensor", "nib.nifti1.Nifti1Image", "sitk.Image"],
        result_all: bool = True,
        save_group_times: bool | None = None,
        log_times: bool | None = None,
        verbose: bool | None = None,
    ) -> dict[str, PanopticaResult]:
```

How about this?